fontforge/namelist.c

```diff
@@ -202,7 +202,7 @@ const char *StdGlyphName(char *buffer, int uni,enum uni_interp interp,NameList *
 	}
     }
     if ( name==NULL ) {
-	if ( uni>=0x10000 )
+	if ( uni>=0x10000 || uni < 0 )
 	    sprintf( buffer, "u%04X", uni);
```

Comment:
If `uni < 0` then neither of the two branches here make any sense (they are checking whether to use `u` or `uni` prefix, but both expect +ve number). I think the code should return either NULL or if caller don’t check for that, some nonsensical name as `.none` (not `.null` because there is standard glyph with that name).

Reply:
Would that be really much more sensible than what the code does for MAXINT? Is there any justification for FontForge having made this a signed value in the first place (other than convenience)? 

I am happy to make U+10FFFF the upper bound if my reviewers are. I'm less happy to implement an arbitrary bound at MAXINT+1.

Reply:
Making the upper bound the maximum valid Unicode characters makes sense as well.

Reply:
Hmm, actually, adding checks at all of those points means adding error paths in some places that currently don't have them. That just seems like an intrusive change likely to create bugs. 

My current inclination is to add `StdGlyphNameBoundsCheck()` with the proposed behavior to centralize the logic and to leave StdGlyphName as I have altered it. I'll shift any calls made from the Python interface to use the new function.

Reply:
Make `StdGlyphName()` call the new `StdGlyphNameBoundsCheck()` and check its result and return a dummy glyph name if it returned `NULL`. Thinking again, I think the new function should return `NULL` for -ve numbers only to be a less disruptive change.

Reply:
I'm not comfortable altering the code to start assigning the same name to multiple glyphs when called in questionable ways. ...BoundsCheck can return NULL for anything outside of 0..0x10FFFF. StdGlyphName can log an error but return a string unique to every int it's called with. 

Your route may be better in absolute terms but it requires more code inspection than I'm interested in doing to fix this problem.

Reply:
> Would that be really much more sensible than what the code does for MAXINT? Is there any justification for FontForge having made this a signed value in the first place (other than convenience)?

The reason for the signed value is to work with .nodef. in some Unicode/utype.c tables.
If you're going to work on this code, you'll want to consider unicode goes to 0..10ffff, but an older ISO has an even larger range, I added comments here: https://github.com/fontforge/fontforge/blob/master/Unicode/ustring.c:699, however, do what you think improves it versus doing nothing at all, since an incremental improvement is preferred over nothing.
Additionally, it's also worth thinking of possible scenarios between int32 and int64 mayhem, so if you can avoid plain int, it may help prevent mystery magic later on.

Reply:
I just did a new push. The integer size shouldn't affect the behavior of `StdGlyphNameBoundsCheck()` as all relevant values are comfortably within the range of either. `StdGlyphName()` will print an error for larger or negative values, but should be fine as long as the buffer passed is big enough.